src/time-picker/time-display.jsx

```diff
@@ -102,40 +104,78 @@ const TimeDisplay = React.createClass({
       },
 
       box: {
-        padding: '16px 0',
+        padding: '14px 0',
         borderTopLeftRadius: 2,
         borderTopRightRadius: 2,
         backgroundColor: this.getTheme().headerColor,
-        color: this.getTheme().textColor,
+        color: 'white',
       },
 
-      text: {
-        color: 'white',
-        opacity: 0.7,
+      affix: {
+        position: 'relative',
+        lineHeight: '17px',
+        height: 17,
+        fontSize: '17px',
+      },
+
+      affixTop: {
+        position: 'absolute',
+        top: '-20px',
+        left: 0,
       },
 
-      hour: {},
+      clickable: {
+        cursor: 'pointer',
+      },
 
-      minute: {},
+      inactive: {
+        opacity: 0.7,
+      },
     };
 
     let [hour, min] = this.sanitizeTime();
 
-    styles[mode].opacity = 1.0;
+    let buttons = null;
+    let buttonsLeft = null;
+    if (this.props.format === 'ampm') {
+      buttons = (
+        <span style={this.prepareStyles(styles.affix)}>
+          <span
+            style={this.prepareStyles(styles.clickable, affix === 'pm' ? {} : styles.inactive)}
+            onTouchTap={() => this.props.onSelectAffix('pm')}>
+            {"PM"}
+          </span>
+          <span
+            style={this.prepareStyles(styles.affixTop, styles.clickable, affix === 'am' ? {} : styles.inactive)}
+            onTouchTap={() => this.props.onSelectAffix('am')}>
+            {"AM"}
+          </span>
+        </span>
+      );
+
+      buttonsLeft = (<span style={this.prepareStyles(styles.affix, {opacity: 0.0})}>{"AM"}</span>);
```

Comment:
So we are using an invisible component to position another component.
That's sounds wrong.